https://gitlab.com/sane-project/backends/-/commit/12560890a6e298091bd63b8093a35604416eb92a

From 12560890a6e298091bd63b8093a35604416eb92a Mon Sep 17 00:00:00 2001
From: David Ward <david.ward@gatech.edu>
Date: Thu, 21 Apr 2022 23:37:33 -0400
Subject: [PATCH] Fix header file used for poll()

POSIX specifies the header to include is <poll.h>, not <sys/poll.h>.
This results in a compiler warning with musl libc (on Alpine Linux).
--- a/configure.ac
+++ b/configure.ac
@@ -202,7 +202,7 @@ AC_HEADER_STDC
 AC_CHECK_HEADERS(fcntl.h unistd.h libc.h sys/dsreq.h sys/select.h \
     sys/time.h sys/shm.h sys/ipc.h sys/scanio.h os2.h \
     sys/socket.h sys/io.h sys/hw.h sys/types.h linux/ppdev.h \
-    dev/ppbus/ppi.h machine/cpufunc.h sys/sem.h sys/poll.h \
+    dev/ppbus/ppi.h machine/cpufunc.h sys/sem.h poll.h \
     windows.h be/kernel/OS.h limits.h sys/ioctl.h asm/types.h\
     netinet/in.h tiffio.h ifaddrs.h pwd.h getopt.h)
 AC_CHECK_HEADERS([asm/io.h],,,[#include <sys/types.h>])
--- a/frontend/saned.c
+++ b/frontend/saned.c
@@ -84,8 +84,8 @@
 
 #include "lgetopt.h"
 
-#if defined(HAVE_SYS_POLL_H) && defined(HAVE_POLL)
-# include <sys/poll.h>
+#if defined(HAVE_POLL_H) && defined(HAVE_POLL)
+# include <poll.h>
 #else
 /*
  * This replacement poll() using select() is only designed to cover
GitLab
